Include/internal/pycore_signal.h

```diff
@@ -29,6 +42,77 @@ extern "C" {
 #  define Py_NSIG 64               // Use a reasonable default value
 #endif
 
+#if defined(MS_WINDOWS) && defined(SOCKET)
+#  define INVALID_FD ((SOCKET)-1)
+#else
+#  define INVALID_FD (-1)
+#endif
+
+struct _signals_runtime_state {
+    volatile struct {
+        _Py_atomic_int tripped;
+        /* func is atomic to ensure that PyErr_SetInterrupt is async-signal-safe
+         * (even though it would probably be otherwise, anyway).
+         */
+        _Py_atomic_address func;
+    } handlers[Py_NSIG];
+
+    volatile struct {
+#ifdef MS_WINDOWS
+#  ifdef SOCKET
+        SOCKET fd;
+#  else
+        // <winsock2.h> wasn't included already,
+        // we use something compatible with SOCKET.
+        int fd;
+#  endif
+#elif defined(__VXWORKS__)
+        int fd;
+#else
+        sig_atomic_t fd;
+#endif
+
+        int warn_on_full_buffer;
+#ifdef MS_WINDOWS
+        int use_send;
+#endif
+    } wakeup;
+
+    /* Speed up sigcheck() when none tripped */
+    _Py_atomic_int is_tripped;
+
+    /* These objects necessarily belong to the main interpreter. */
+    PyObject *default_handler;
+    PyObject *ignore_handler;
+#ifdef MS_WINDOWS
+#  ifdef HANDLE
+    HANDLE sigint_event;
+#  else
+    // <windows.h> wasn't included already,
+    // we use something compatible with HANDLE.
+    void *sigint_event;
+#  endif
+#endif
```

Comment:
@zooba, if I understood you correctly, `void *` is an appropriate substitute for `HANDLE`.  I'm a bit more comfortable about that.